File: app/src/main/java/com/verbatoria/presentation/late_send/view/adapter/LateReportViewHolder.java
package com.verbatoria.presentation.late_send.view.adapter;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.remnev.verbatoriamini.R;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * View Holder для отображения ячейки позднего отчета
 *
 * @author nikitaremnev
 */
public class LateReportViewHolder extends RecyclerView.ViewHolder {

    private View mRootView;

    @BindView(R.id.field_image_view)
    public ImageView mFieldImageView;

    @BindView(R.id.field_title)
    public TextView mTitle;

    @BindView(R.id.field_subtitle)
    public TextView mSubtitle;

    LateReportViewHolder(View itemView) {
        super(itemView);
        ButterKnife.bind(this, itemView);
        mRootView = itemView;
    }

    void setTitle(String title) {
        mTitle.setText(title);
    }

    void setSubtitle(String subtitle) {
        mSubtitle.setText(subtitle);
    }

    void showImage() {
        mFieldImageView.setImageResource(R.drawable.ic_late_report);
    }

    void setOnClickListener(View.OnClickListener onClickListener) {
        mRootView.setOnClickListener(onClickListener);
    }

}

File: app/src/main/java/com/verbatoria/presentation/late_send/item/LateReportViewHolder.kt
package com.verbatoria.presentation.late_send.item

import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import com.remnev.verbatoria.R

/**
 * @author n.remnev
 */

interface LateReportViewHolder {

    fun setTitle(title: String)

    fun setSubtitle(subtitle: String)

    interface Callback {

        fun onLateReportClicked(position: Int)

    }

}

class LateReportViewHolderImpl(
    rootView: View,
    callback: LateReportViewHolder.Callback
) : RecyclerView.ViewHolder(rootView), LateReportViewHolder {

    private val titleTextView: TextView = rootView.findViewById(R.id.field_title)
    private val subtitleTextView: TextView = rootView.findViewById(R.id.field_title)
    private val logoImageView: ImageView = rootView.findViewById(R.id.field_image_view)

    init {
        logoImageView.setImageResource(R.drawable.ic_late_report)
        rootView.setOnClickListener {
            callback.onLateReportClicked(adapterPosition)
        }
    }

    //region LateReportViewHolder

    override fun setTitle(title: String) {
        titleTextView.text = title
    }

    override fun setSubtitle(subtitle: String) {
        subtitleTextView.text = subtitle
    }

    //endregion

}